pulsar/test_helper.go

```diff
@@ -164,3 +173,55 @@ func topicPath(topic string) string {
 	}
 	return tn.Name
 }
+
+func createClient(t *testing.T) Client {
+	client, err := NewClient(ClientOptions{
+		URL: serviceURL,
+	})
+	assert.Nil(t, err)
+
+	return client
+}
+
+func createConsumer(t *testing.T, client Client, options ConsumerOptions) Consumer {
+	consumer, err := client.Subscribe(options)
+	assert.Nil(t, err)
+
+	return consumer
+}
+
+func createProducer(t *testing.T, client Client, options ProducerOptions) Producer {
+	producer, err := client.CreateProducer(options)
+	assert.Nil(t, err)
+	assert.Equal(t, options.Topic, producer.Topic())
+
+	return producer
+}
+
+func send(t *testing.T, producer Producer, ctx context.Context, times int, message *ProducerMessage) {
```

Comment:
Check the consumer_regex_test genMessages https://github.com/apache/pulsar-client-go/blob/master/pulsar/consumer_regex_test.go#L361 this might be a way to make it more flexible.

Reply:
OK.